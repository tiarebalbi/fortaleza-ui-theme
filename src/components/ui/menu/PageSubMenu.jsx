// @flow
import * as React from 'react';
import type { PageSubMenuType } from '../../../types';

import Button from '../Button';
import PageSubMenuStyle from './styles/PageSubMenuStyle';

export default class PageSubMenu extends React.PureComponent<PageSubMenuType> {
  render() {
    const { title, description, link, children } = this.props;

    return (
      <PageSubMenuStyle>
        <div className="title-content">
          <h1>{title}</h1>
          {description && <p>{description}</p>}
          {link && (
            <Button
              className="btn-sub-menu-page"
              palette="secondary"
              href={link.href}
              icon={link.icon}
            >
              {link.name}
            </Button>
          )}
        </div>
        <div className="options">{children}</div>
      </PageSubMenuStyle>
    );
  }
}
