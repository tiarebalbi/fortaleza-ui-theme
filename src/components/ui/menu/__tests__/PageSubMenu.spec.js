import React from 'react';
import { TopNavMenu } from '../index';
import { mount } from 'enzyme';
import PageSubMenu from '../PageSubMenu';

const setup = (props = {}) => {
  return mount(<PageSubMenu {...props} />);
};

describe('ui/PageSubMenu', () => {
  const wrapper = setup({ title: 'Home' });

  it('should be stateful', () => {
    expect(wrapper.instance()).not.toBeNull();
  });

  it('should be able to match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should be able to render description', () => {
    wrapper.setProps({
      description: 'Sample Description',
    });

    expect(wrapper.find('p')).toHaveText('Sample Description');
  });

  it('should be able to render link', () => {
    wrapper.setProps({
      link: {
        name: 'Custom Button',
        link: '/',
      },
    });

    expect(wrapper.find('a.btn-sub-menu-page span')).toHaveText(
      'Custom Button',
    );
  });
});
