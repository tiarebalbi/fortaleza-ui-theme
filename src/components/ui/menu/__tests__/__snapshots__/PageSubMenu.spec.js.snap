// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/PageSubMenu should be able to match snapshot 1`] = `
<PageSubMenu
  title="Home"
>
  <styled.div>
    <div
      className="sc-gzVnrw bIIlHi"
    >
      <div
        className="title-content"
      >
        <h1>
          Home
        </h1>
      </div>
      <div
        className="options"
      />
    </div>
  </styled.div>
</PageSubMenu>
`;
